sdk/cosmos/azure-cosmos/test/query_tests.py

```diff
@@ -214,6 +218,177 @@ def _MockExecuteFunction(self, function, *args, **kwargs):
         self.count += 1
         return self.OriginalExecuteFunction(function, *args, **kwargs)
 
+    def test_get_query_plan_through_gateway(self):
+        created_collection = self.config.create_multi_partition_collection_with_custom_pk_if_not_exist(self.client)
+        self._validate_query_plan("Select top 10 value count(c.id) from c", created_collection.container_link, 10, [], ['Count'], True, None, None, "None")
+        self._validate_query_plan("Select * from c order by c._ts offset 5 limit 10", created_collection.container_link, None, ['Ascending'], [], False, 5, 10, "None")
+        self._validate_query_plan("Select distinct value c.id from c order by c.id", created_collection.container_link, None, ['Ascending'], [], True, None, None, "Ordered")
+
+    def _validate_query_plan(self, query, container_link, top, order_by, aggregate, select_value, offset, limit, distinct):
+        query_plan_dict = self.client.client_connection._GetQueryPlanThroughGateway(query, container_link)
+        query_execution_info = _PartitionedQueryExecutionInfo(query_plan_dict)
+        self.assertTrue(query_execution_info.has_rewritten_query())
+        self.assertEquals(query_execution_info.has_distinct_type(), distinct != "None")
+        self.assertEquals(query_execution_info.get_distinct_type(), distinct)
+        self.assertEquals(query_execution_info.has_top(), top is not None)
+        self.assertEquals(query_execution_info.get_top(), top)
+        self.assertEquals(query_execution_info.has_order_by(), len(order_by) > 0)
+        self.assertListEqual(query_execution_info.get_order_by(), order_by)
+        self.assertEquals(query_execution_info.has_aggregates(), len(aggregate) > 0)
+        self.assertListEqual(query_execution_info.get_aggregates(), aggregate)
+        self.assertEquals(query_execution_info.has_select_value(), select_value)
+        self.assertEquals(query_execution_info.has_offset(), offset is not None)
+        self.assertEquals(query_execution_info.get_offset(), offset)
+        self.assertEquals(query_execution_info.has_limit(), limit is not None)
+        self.assertEquals(query_execution_info.get_limit(), limit)
+
+    def test_unsupported_queries(self):
+        created_collection = self.config.create_multi_partition_collection_with_custom_pk_if_not_exist(self.client)
+        queries = ['SELECT COUNT(1) FROM c', 'SELECT COUNT(1) + 5 FROM c', 'SELECT COUNT(1) + SUM(c) FROM c']
```

Comment:
We still need to support "SELECT COUNT(1) FROM c" in the v2 SDK. There should be some check that sends it down the other pipeline.

Reply:
correction. we need to support it for single partition collections.

Reply:
yes, this is only for multi partition collections, with enableCrossPartitionuery set to True